### ISSUE TITLE
globalValue raises error from mpi4py

### ISSUE THREAD

**Issue by guyer** [usnistgov/fipy#491]
As reported [on the mailing list](http://permalink.gmane.org/gmane.comp.python.fipy/4072), calling `var.globalValue` in parallel rasies
```python
(parallel)N103898:fipy guyer$ mpirun -np 2 python test.py --trilinos
hello from 0 out of 2
hello from 1 out of 2
Traceback (most recent call last):
  File "test.py", line 6, in <module>
    print 'hello from',fp.tools.parallel.procID,'out of',fp.tools.parallel.Nproc,p.globalValue
Traceback (most recent call last):
  File "test.py", line 6, in <module>
    print 'hello from',fp.tools.parallel.procID,'out of',fp.tools.parallel.Nproc,p.globalValue
  File "/Users/guyer/Documents/research/FiPy/fipy/fipy/variables/cellVariable.py", line 162, in globalValue
  File "/Users/guyer/Documents/research/FiPy/fipy/fipy/variables/cellVariable.py", line 162, in globalValue
    self.mesh._globalNonOverlappingCellIDs)
    self.mesh._globalNonOverlappingCellIDs)
  File "/Users/guyer/Documents/research/FiPy/fipy/fipy/variables/meshVariable.py", line 152, in _getGlobalValue
  File "/Users/guyer/Documents/research/FiPy/fipy/fipy/variables/meshVariable.py", line 152, in _getGlobalValue
    globalIDs = numerix.concatenate(self.mesh.communicator.allgather(globalIDs))
  File "/Users/guyer/Documents/research/FiPy/fipy/fipy/tools/comms/mpi4pyCommWrapper.py", line 75, in allgather
    globalIDs = numerix.concatenate(self.mesh.communicator.allgather(globalIDs))
  File "/Users/guyer/Documents/research/FiPy/fipy/fipy/tools/comms/mpi4pyCommWrapper.py", line 75, in allgather
    return self.mpi4py_comm.allgather(sendobj=sendobj, recvobj=recvobj)
  File "MPI/Comm.pyx", line 1288, in mpi4py.MPI.Comm.allgather (src/mpi4py.MPI.c:109141)
    return self.mpi4py_comm.allgather(sendobj=sendobj, recvobj=recvobj)
  File "MPI/Comm.pyx", line 1288, in mpi4py.MPI.Comm.allgather (src/mpi4py.MPI.c:109141)
TypeError: allgather() got an unexpected keyword argument 'recvobj'
TypeError: allgather() got an unexpected keyword argument 'recvobj'
```

The `recvobj` argument was [removed in May 2014](https://bitbucket.org/mpi4py/mpi4py/commits/3d8503a11d320dd1c3030ec0dbce95f63b0ba602) and we apparently never noticed. It looks like we can safely delete it from our code as it was apparently never used for anything.

**Comment by tkphd** [comment]
The mpi4py commit mentions that the receive object is no longer needed for the lower-case form of the commands. Browsing the full source shows that the upper-case commands retain both the send and receive objects. To avoid deviating too far from the MPI standard, I'd like to suggest changing the case (Allgather instead of allgather), rather than dropping buffers, in our mpi4pyCommWrapper.py.

**Comment by tkphd** [comment]
Looking into the rest of the FiPy source, we're already calling allgather(sendobj) in several places, and only calling allgather(sendobj, recvobj) in mpi4pyCommWrapper.py. To preserve the existing function calls (all of which are lower-case), removing the recvobj argument appears to be the right call after all.
